code/game/mecha/mecha_decal_kits.dm

```diff
@@ -0,0 +1,561 @@
+        var/datum/mecha/mecha_decal/D = locate(hrefstring)
+        D.decal_colour = input("Select the new decal colour:", "Decal Colour", D.decal_colour) as color|null
+        colour_overlay(D.decal_colour)
+        openUI(current_mecha, usr)
+
+    if(href_list["decalstrip"])
+        if(current_mecha.occupant)
+            to_chat(usr, "<span class='warning'>The exosuit must be unoccupied before you can paint it!</span>")
+            return
+        if(current_mecha.Adjacent(usr) && do_after_once(usr, 15, target = current_mecha))
+            var/hrefstring = href_list["decalstrip"]
+            var/datum/mecha/mecha_decal/D = locate(hrefstring)
+            current_mecha.strip_decal(D)
+            qdel(D)
+            to_chat(usr, "<span class='notice'>You carefully strip off the decal.</span>")
+            playsound(loc, spray_sound, 50, 1, -6)
+            openUI(current_mecha, usr)
+
+    if(href_list["decalapply"])
+        if(current_mecha.occupant)
+            to_chat(usr, "<span class='warning'>The exosuit must be unoccupied before you can paint it!</span>")
+            return
+        var/hrefstring = href_list["decalapply"]
+        var/obj/item/mecha_decal_container/D = locate(hrefstring)
+        if(current_mecha.Adjacent(usr) && do_after_once(usr, 15, target = current_mecha))
+            if(D.install_on_mecha(current_mecha))
+                to_chat(usr, "<span class='notice'>You spray the exosuit down as tiny manipulators handle the small details.</span>")
+                playsound(loc, spray_sound, 50, 1, -6)
+                openUI(current_mecha, usr)
+
+    if(href_list["decaldelete"])
+        var/hrefstring = href_list["decaldelete"]
+        var/obj/item/mecha_decal_container/D = locate(hrefstring)
+        delete_decal_container(D)
+        openUI(current_mecha, usr)
+
+// Emagging adds the syndicate logo to the paint gun but otherwise does nothing. Pointless badassery?
+/obj/item/mecha_paint_gun/emag_act()
+    if(!emagged)
+        available_decal_containers.Add(new/obj/item/mecha_decal_container/syndicate_logo)
+        visible_message("[src] sparks briefly.")
+        emagged = TRUE
+
+// Creates and spawns a new exosuit pattern on the ground featuring the current exosuit design. 30 second cooldown.
+/obj/item/mecha_paint_gun/proc/save_pattern(obj/mecha/M, list/datum/mecha/mecha_decal/decal_list, var/pattern_name) 
+    if(world.time >= next_pattern_at)
+        var/obj/item/mecha_decal_container/D = new()
+        D.name = "custom exosuit decal pattern ([pattern_name])"
+        D.decal_name = "CUSTOM: [pattern_name]"
+        D.desc = "A custom exosuit design that also overwrites the base coat of the target exosuit."
+        D.decals = list()
+        for(var/datum/mecha/mecha_decal/MD in decal_list)
+            D.decals.Add(MD.clone())
+        D.compatible_mecha = list(M.type)     // Only the exosuit this pattern was made for is compatible with a custom save.
+        D.loc = usr.loc
```

Comment:
Don't do direct loc changes please. Use `forceMove` for that. Or better in this case. Give the new the location as first parameter